tests/test_similarityrecommender.py

```diff
@@ -0,0 +1,194 @@
+import scipy
+from taar.recommenders.similarity_recommender import SimilarityRecommender
+
+FAKE_DONOR_DATA = [
+    {"activeAddons":
+        ["{test-guid-1}", "{test-guid-2}", "{test-guid-3}", "{test-guid-4}"],
+     "geoCity": "nowhere-us",
+     "subsessionLength": 6500,
+     "locale": "en-US",
+     "os": "mac",
+     "bookmark_count": 77,
+     "tab_open_count": 32,
+     "total_uri": 87,
+     "unique_tlds": 65
+     },
+    {"activeAddons":
+        ["{test-guid-5}", "{test-guid-6}", "{test-guid-7}", "{test-guid-8}"],
+     "geoCity": "pompei-it",
+     "subsessionLength": 67832,
+     "locale": "it-IT",
+     "os": "Linux",
+     "bookmark_count": 8166,
+     "tab_open_count": 232,
+     "total_uri": 42342,
+     "unique_tlds": 1203
+     },
+    {"activeAddons":
+        ["{test-guid-9}", "{test-guid-10}", "{test-guid-11}", "{test-guid-12}"],
+     "geoCity": "brasilia-br",
+     "subsessionLength": 1011,
+     "locale": "br-PT",
+     "os": "windows",
+     "bookmark_count": 17,
+     "tab_open_count": 3,
+     "total_uri": 432,
+     "unique_tlds": 10
+     }
+]
+
+
+def generate_fake_lr_curves(num_elements):
+    lr_index = list(np.linspace(0, 10, num_elements))
+    numerator_density = [scipy.stats.norm.pdf(float(i), 0.5, 0.5) for i in lr_index]
+    denominator_density = [scipy.stats.norm.pdf(float(i), 5.0, 1.5) for i in lr_index]
+
+    # JSON conversion breaks the tuple, so lists are needed.
+    p_pairs = [list(a) for a in zip(numerator_density, denominator_density)]
+    return [list(a) for a in zip(lr_index, p_pairs)]
+
+
+def generate_a_fake_taar_client():
+    return {
+        "client_id": "test-client-001",
+        "activeAddons": [],
+        "geoCity": "rio-br",
+        "subsessionLength": 911,
+        "locale": "br-PT",
+        "os": "windows",
+        "bookmark_count": 57,
+        "tab_open_count": 4,
+        "total_uri": 222,
+        "unique_tlds": 21
+    }
+
+
+def instantiate_mocked_s3_bucket():
+    bucket = 'test-bucket'
+    conn = boto3.resource('s3', region_name='us-west-2')
+    conn.create_bucket(Bucket=bucket)
+
+    # Write the fake addon donor data to the mocked S3.
+    conn.Object(bucket, key='taar/legacy/addon_donors.json').put(Body=json.dumps(FAKE_DONOR_DATA))
+    # Write the fake lr curves data to the mocked S3.
+    fake_lrs = generate_fake_lr_curves(1000)
+    conn.Object(bucket, key='taar/legacy/test/lr_curves.json').put(Body=json.dumps(fake_lrs))
+
+
+@mock_s3
+def test_can_recommend():
+    # Create a new instance of a SimilarityRecommender.
+    r = SimilarityRecommender()
+
+    # Test that we can't recommend if we have not enough client info.
+    assert not r.can_recommend({})
+
+    # Test that we can recommend for a normal client.
+    assert r.can_recommend(generate_a_fake_taar_client())
+
+    # Check that we can not recommend if any required client field is missing.
+    required_fields = ["geoCity", "subsessionLength", "locale", "os", "bookmark_count", "tab_open_count",
```

Comment:
Let's just combine the lists exported from the similarity modules instead of defining a new one.